python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -167,6 +169,49 @@ def decorated(self, *args, **kwargs):
     return decorated
 
 
+class RegistryFlusher(threading.Thread):
```

Comment:
It would be nice to see it do this, but I suspect (for safety at least) we want to handle this entirely within the registries to make sure it is not mistakenly shutdown out of order with the registry in the shutdown of Ganga.